p32/src/bank.rs

```diff
@@ -0,0 +1,248 @@
+        Self {
+            name,
+            credit_line,
+            balance,
+        }
+    }
+}
+
+impl Bank {
+    pub fn new(name: String, credit_interest: u64, debit_interest: u64) -> Self {
+        Self {
+            name,
+            users: Vec::new(),
+            credit_interest,
+            debit_interest,
+        }
+    }
+
+    pub fn add_user(&mut self, user: User) {
+        self.users.push(user);
+    }
+
+    pub fn calc_balance(&self) -> (u64, u64) {
+        let mut liabilities = 0;
+        let mut assets = 0;
+
+        for user in &self.users {
+            if user.balance > 0 {
+                liabilities += user.balance as u64;
+            } else {
+                assets += (-user.balance) as u64;
+            }
+        }
+
+        (liabilities, assets)
+    }
+
+    pub fn transfer_funds(
+        &mut self,
+        from_name: &str,
+        to_name: &str,
+        amount: u64,
+    ) -> Result<(), TransferError> {
+        let amount_i64 = amount as i64;
+
+        // Find user indices first
+        let from_idx = self
+            .users
+            .iter()
+            .position(|u| u.name == from_name)
+            .ok_or(TransferError::UserNotFound(from_name.to_string()))?;
+
+        let to_idx = self
+            .users
+            .iter()
+            .position(|u| u.name == to_name)
+            .ok_or(TransferError::UserNotFound(to_name.to_string()))?;
+
+        // Check if transfer is possible
+        if self.users[from_idx].balance < amount_i64 {
+            return Err(TransferError::InsufficientFunds(from_name.to_string()));
+        }
+
+        if (self.users[from_idx].balance - amount_i64).unsigned_abs()
+            > self.users[from_idx].credit_line
+        {
+            return Err(TransferError::CreditLimitExceeded(from_name.to_string()));
+        }
+
+        if (self.users[to_idx].balance + amount_i64).unsigned_abs() > self.users[to_idx].credit_line
+        {
+            return Err(TransferError::CreditLimitExceeded(to_name.to_string()));
+        }
+
+        // Execute transfer
+        self.users[from_idx].balance -= amount_i64;
+        self.users[to_idx].balance += amount_i64;
+
+        Ok(())
+    }
+
+    pub fn accrue_interest(&mut self) {
+        use std::cmp::Ordering;
+
+        for user in &mut self.users {
+            match user.balance.cmp(&0) {
+                Ordering::Greater => {
+                    // Debit interest with rounding
+                    let interest = (user.balance * self.debit_interest as i64 + 5000) / 10000;
+                    user.balance += interest;
+                }
+                Ordering::Less => {
+                    // Credit interest with rounding
+                    let abs_balance = (-user.balance) as u64;
+                    let interest = ((abs_balance * self.credit_interest + 5000) / 10000) as i64;
+                    user.balance -= interest;
```

Comment:
Because you know that the balance is negative, you can simplify it and write `user.balance += user.balance * self.credit_interest / 10_000;`.